lang/csharp/src/apache/main/Util/TimeMicrosecond.cs

```diff
@@ -50,17 +51,16 @@ public override object ConvertToBaseValue(object logicalValue, LogicalSchema sch
         {
             var time = (TimeSpan)logicalValue;
 
-            if (time > _maxTime)
+            if (time >= _maxTime)
                 throw new ArgumentOutOfRangeException(nameof(logicalValue), "A 'time-micros' value can only have the range '00:00:00' to '23:59:59'.");
```

Comment:
The message of the exception should be corrected for the exclusive upper bound (because e.g. the time-micros value 23:59:59 and 999 milliseconds would be valid). How about: A 'time-micros' value has to be at least '00:00:00' and less than '24:00:00'.

Reply:
I am always paranoid when it comes to changing existing exception messages. My concern is that the ned user is using a string match to capture avro extensions, since there are very exception types, and the exceptions are differentiated by their message.

Additionally from a human reading point of the message is correct(ish). When I read it as it is now, my expectation is that it is less than 24 hours.

Reply:
The bigger issue is that `ConvertToLogicalValue` does not check if the base value (long) is >=0 and less than 1 day. IMO it should throw the same exception. However this change could be a drastic addition, because the code would start to throw exceptions where it did not throw before. (even it should). I would ugnore that issue for now.

Reply:
In regards to the exception message: I am not sure that it is correct enough because it is directly in contradiction to the resolution offered by the logical type. I understand the worry about users string matching the exception message - but should such a approach be encouraged? Maybe it's better to continue to improve exception messages and just make it explicit in the release notes. If you decide to change the message, another way could be to just write the lower bound as 00:00:00:000000 and the inclusive upper bound as 23:59:59:999999.

For the lower bound - you are right, I also saw this but got interrupted. I think it should be fine to throw an exception when the base value is < 0 - at least the Avro 1.11.0 specification is clear about the lower bound that implicitly excludes values < 0.

Reply:
The changes to you mention above makes sense. Lets mark this PR as 1.12 version and the dependent PR as well. In that case I feel more comfortable to introduce potentially painful changes to end the end user. E.g changed exception message or additional exceptions in case of incorrect values.

Other option to keep the scope of ticket as os and let's open a ticket for the changes mentioned above and we will mark that ticket as as 12.0.

Reply:
I think I would prefer the second option as I am looking forward very much to use default values with logical types :-)

Reply:
Ok. I will make the ticket when this is approved and merged and will make the new ticket marked as 12.0. It is not a complicated one so it should be fairly trivial including the tests. If everyone agrees with the risks of course we can merge it sooner and make it part of a minor release instead of waiting for 12.0

Reply:
Thank you very much!

Reply:
@anekdoti Plz approve the PR if you are ok with the changes.